### PR TITLE
ZCS-14481: Adds utils/monitor_soap.sh script.

### PR DESCRIPTION
**Pull request by GaneshAnarse**
**ZCS-14481: Adds utils/monitor_soap.sh script.**
* This script will monitor the SOAP execution.
* It will send an email notification to provided email address once SOAP execution is complete.

Note: Email Notifications for ubuntu systems is not working correctly at the moment.
This will be addressed later in further tickets related to SOAP automation using jenkins.

### CODE REVIEW

File: utils/monitor_soap.sh [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    check_staf
+    set_staf
+    set_properties
+    monitor_soap_run &
+}
+
+function monitor_soap_run {
+    soap_run_pid="$(run_tests)"
+    echo "Monitoring the SOAP run."
+    echo "PID: $soap_pid"
```
Comment (umagmrit) [inline utils/monitor_soap.sh:91]: `$soap_pid ` or `$soap_run_pid` ?

File: utils/monitor_soap.sh [hunk 2]
```diff
@@ -0,0 +1,250 @@
+#!/bin/bash
+# 
+# This Script can trigger the soap run and monitor it until completion.
```
Comment (umagmrit) [inline utils/monitor_soap.sh:3]: can you rename this script name as it is capable of trigger the soap run and monitor it ?

File: utils/monitor_soap.sh (hunk 1 above)
Comment (GaneshAnarse) [reply utils/monitor_soap.sh]: Corrected the variable name!

File: utils/monitor_soap.sh (hunk 2 above)
Comment (GaneshAnarse) [reply utils/monitor_soap.sh:3]: It can do both things sir!
Can I name it as `run_and_monitor_soap.sh`??

Comment (umagmrit) [reply utils/monitor_soap.sh:3]: yes

File: utils/monitor_soap.sh [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    cat <<EOF > ./email.txt
+    To: $NOTIFY_EMAIL
+    Subject: "SOAP: Tests execution started at $(date)"
+
+    Status of the SOAP execution on $HOSTNAME
+    ------------------------------------------------------------
+    Host: $HOSTNAME
+    Test Path: $TEST_PATH
+    Test Suite: $TEST_SUITE
+    SOAP test results are now available at: $dir_soaprun
```
Comment (umagmrit) [inline utils/monitor_soap.sh:112]: can we mail the SOAP results as well ?

File: utils/monitor_soap.sh [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    sudo sed -i "s|localhost|$HOSTNAME|g" $dir_qa/soapvalidator/conf/global.properties
+    sudo sed -i "s|@zimbra\.com|@$domain_name|g" $dir_qa/soapvalidator/conf/global.properties
+    sudo sed -i "s|defaultdomain\.name=.*|defaultdomain.name=$domain_name|g" $dir_qa/soapvalidator/conf/global.properties
+    sudo sed -i "s|admin\.password=.*|admin.password=$ADMIN_PASS|g" $dir_qa/soapvalidator/conf/global.properties
+    sudo sed -i "s|admin\.user=.*|admin.password=$ADMIN_USER|g" $dir_qa/soapvalidator/conf/global.properties
+}
+
+function run_tests {
+    # Execute SOAP tests on the server
+    STAF LOCAL soap \
```
Comment (umagmrit) [inline utils/monitor_soap.sh:240]: I think for full test suite we need to use `nohup` to run it in the background.

File: utils/monitor_soap.sh [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+
+function main {
+    set -x
+    dir_bin="$(cd "$(dirname "$0")" && pwd)"
+    dir_qa="/opt/qa"
+    dir_staf="/usr/local/staf"
+    dir_soaprun="/opt/soaprun/"
+    notify_email="ganesh.anarse@syancor.com"
+    HOSTNAME="$(hostname -f)"
+    TEST_PATH="$dir_qa/soapvalidator/data/soapvalidator/Admin/Auth/AdminAuth-Basic.xml"
```
[the remark this answers is not in the document]
Comment (GaneshAnarse) [reply utils/monitor_soap.sh:29]: Yes sir, But for now I am keep one test for the sake of testing!

File: utils/monitor_soap.sh (hunk 3 above)
Comment (GaneshAnarse) [reply utils/monitor_soap.sh:112]: Big file attachments do not get emailed correctly.
We will have to analyse the results with some function to consolidate it and email to the user.
I think we should take it up in some other ticket sir!

File: utils/monitor_soap.sh (hunk 4 above)
Comment (GaneshAnarse) [reply utils/monitor_soap.sh]: Added nohup command to run tests in the backgroud!

File: utils/monitor_soap.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (umagmrit) [reply utils/monitor_soap.sh:29]: Have you triggered build with these changes ?

**Comment by GaneshAnarse** [comment]
The SOAP execution is now getting triggered and reported successfully, checkout - 
http://144.24.97.112:8080/job/deploy-dev/91/console

File: utils/monitor_soap.sh (hunk 4 above)
Comment (GaneshAnarse) [reply utils/monitor_soap.sh]: Hi @umagmrit sir, By remotely triggering the execution and monitor script with `&` seems to be working well here.
Checkout - http://144.24.97.112:8080/job/deploy-dev/91/console

File: utils/monitor_soap.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (GaneshAnarse) [reply utils/monitor_soap.sh:28]: Please checkout - http://144.24.97.112:8080/job/deploy-dev/91/console

**Pull request merged**